homeassistant/components/geofency/device_tracker.py

```diff
@@ -38,7 +38,9 @@ def _receive_data(device, gps, location_name, attributes):
     dev_reg = dr.async_get(hass)
     dev_ids = {
         identifier[1]
-        for device in dev_reg.devices.values()
+        for device in dev_reg.devices.get_devices_for_config_entry_id(
+            config_entry.entry_id
+        )
         for identifier in device.identifiers
         if identifier[0] == GF_DOMAIN
```

Comment:
```suggestion
```
Won't all DeviceEntries returned by `get_devices_for_config_entry_id(config_entry.entry_id)` have this identifier?

Reply:
dfe54b9e5f7eddf3749b3bdfb1422a9f40f80b7c